build-support/bin/release.sh

```diff
@@ -558,8 +558,9 @@ function publish_packages() {
 
   activate_twine
   trap deactivate RETURN
-  twine upload --sign "${DEPLOY_WHEEL_DIR}"/*.whl
-  twine upload --sign "${DEPLOY_SDIST_DIR}"/*.tar.gz
+
+  twine upload --sign-with=get_pgp_keyid "${DEPLOY_WHEEL_DIR}"/*.whl
```

Comment:
This needs backticks or `$()` surrounding for `get_pgp_keyid` to be evaluated. I'll send a follow-up with the fix.

Reply:
Actually, the argument to --sign-with is the gpg program to use, so this looks like the wrong option. I think this should be `twine upload --sign --identity=$(get_pgp_keyid) ...`.

Reply:
OK - yeah, this is borne out by comparing https://pypi.python.org/pypi/pantsbuild.pants/1.4.0.dev16 with https://pypi.python.org/pypi/pantsbuild.pants/1.4.0.dev17. The dev16 has pgp sigs in addition to md5 digests on pypi, the dev17 just has md5s.

Reply:
Hmm - so instead of using my pgp key like it seemed, it just skipped using PGP keys entirely. It did ask for my key at least once, and it changed from using the wrong one to using the git one 🤔 

I am around today, if you want me to followup with the fixes, I can do so

Reply:
I'll give it a spin when this fix hits master. Unfortunately both git tags and pypi uploads are mutable, which means though that I can just re-run on master checked out to the release tag.

Reply:
Hrm, I just realized the current wheel generation scheme doesn't allow for a proper re-release if commits sneak in between release and re-release, since those commits will be represented in a new wheel binary on S3. The fix here will have to come with the dev18 release.

Reply:
... or a dev17 re-release with updated release notes (~3 new commits). It looks like the release also failed to tag.

Reply:
The release.sh script exited cleanly when I ran it last night... I agree a re-release sounds in order